test/unit/uk/gov/hmrc/apinotificationqueue/controller/QueueControllerSpec.scala

```diff
@@ -65,10 +72,20 @@ class QueueControllerSpec extends UnitSpec with MockitoSugar with WithFakeApplic
   }
 
   "GET /message/:id" should {
+    val uuid = java.util.UUID.randomUUID()
+
+    "throw exception if x-client-id not present" in new Setup {
+      intercept[BadRequestException] {
+        controllerUnderTest.get(uuid)(FakeRequest("GET", s"/notification/$uuid"))
+      }
+    }
+
     "return 200" in new Setup {
-      val uuid = java.util.UUID.randomUUID()
-      val result = controllerUnderTest.get(uuid)(FakeRequest("GET", s"/notification/$uuid"))
+      private val payload = "<xml>a</xml>"
+      when(mockQueueService.get("a", uuid)).thenReturn(Future.successful(Some(Notification(uuid, Map("content-type" -> "application/xml"), payload, DateTime.now()))))
+      val result = controllerUnderTest.get(uuid)(FakeRequest("GET", s"/notification/$uuid", Headers("x-client-id" -> "a"), AnyContentAsEmpty))
       status(result) shouldBe Status.OK
+      contentAsString(result) shouldBe (payload)
     }
```

Comment:
there should be an assertion that the headers in the response match those returned by `mockQueueService.get`